internal/operands/common-instancetypes/reconcile.go

```diff
@@ -0,0 +1,169 @@
+package common_instancetypes
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+
+	"k8s.io/apimachinery/pkg/util/yaml"
+	instancetypeapi "kubevirt.io/api/instancetype"
+	instancetypev1alpha2 "kubevirt.io/api/instancetype/v1alpha2"
+	"kubevirt.io/ssp-operator/internal/common"
+	"kubevirt.io/ssp-operator/internal/operands"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Define RBAC rules needed by this operand:
+// +kubebuilder:rbac:groups=instancetype.kubevirt.io,resources=virtualmachineclusterinstancetypes,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=instancetype.kubevirt.io,resources=virtualmachineclusterpreferences,verbs=get;list;watch;create;update;patch;delete
+
+const (
+	operandName                      = "common-instancetypes"
+	operandComponent                 = common.AppComponentTemplating
+	BundleDir                        = "data/common-instancetypes-bundle/"
+	ClusterInstancetypesBundlePrefix = "common-clusterinstancetypes-bundle"
+	ClusterPreferencesBundlePrefix   = "common-clusterpreferences-bundle"
+)
+
+type commonInstancetypes struct {
+	virtualMachineClusterInstancetypes []instancetypev1alpha2.VirtualMachineClusterInstancetype
+	virtualMachineClusterPreferences   []instancetypev1alpha2.VirtualMachineClusterPreference
+}
+
+var _ operands.Operand = &commonInstancetypes{}
+
+func fetchVirtualMachineClusterInstancetypeBundle(bundlePath string) ([]instancetypev1alpha2.VirtualMachineClusterInstancetype, error) {
+	file, err := ioutil.ReadFile(bundlePath)
+	if err != nil {
+		return nil, err
+	}
+	decoder := yaml.NewYAMLOrJSONDecoder(bytes.NewReader(file), 1024)
+	var bundle []instancetypev1alpha2.VirtualMachineClusterInstancetype
+	for {
+		instancetype := instancetypev1alpha2.VirtualMachineClusterInstancetype{}
+		err = decoder.Decode(&instancetype)
+		if err == io.EOF {
+			return bundle, nil
+		}
+		if err != nil {
+			return nil, err
+		}
+		if instancetype.Name != "" {
+			bundle = append(bundle, instancetype)
+		}
+	}
+}
+
+func fetchVirtualMachineClusterPreferencesBundle(bundlePath string) ([]instancetypev1alpha2.VirtualMachineClusterPreference, error) {
+
+	file, err := ioutil.ReadFile(bundlePath)
+	if err != nil {
+		return nil, err
+	}
+	decoder := yaml.NewYAMLOrJSONDecoder(bytes.NewReader(file), 1024)
+	var bundle []instancetypev1alpha2.VirtualMachineClusterPreference
+	for {
+		preference := instancetypev1alpha2.VirtualMachineClusterPreference{}
+		err = decoder.Decode(&preference)
+		if err == io.EOF {
+			return bundle, nil
+		}
+		if err != nil {
+			return nil, err
+		}
+		if preference.Name != "" {
```

Comment:
Is there something else which should be checked to make sure the preference is valid?

Reply:
I've dropped this entirely, it didn't really make sense as the validation webhooks will check all of this for us during submission anyway.